clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/AbstractEnableDisableOpenstackAtomicOperation.groovy

```diff
@@ -54,7 +59,29 @@ abstract class AbstractEnableDisableOpenstackAtomicOperation implements AtomicOp
   Void operate(List priorOutputs) {
     String verb = disable ? 'disable' : 'enable'
     String gerund = disable ? 'Disabling' : 'Enabling'
+
     task.updateStatus phaseName, "Initializing $verb server group operation for $description.serverGroupName in $description.region..."
+    def credentials = description.credentials
+
+    if (credentials.credentials.consulConfig?.enabled) {
+      task.updateStatus phaseName, "$gerund server group in Consul..."
+
+      List<String> instanceIds = provider.getInstanceIdsForStack(description.region, description.serverGroupName)
+      instanceIds.each { String instanceId ->
+        Server instance = provider.getServerInstance(description.region, instanceId)
+        try {
+          EnableDisableConsulInstance.operate(credentials.credentials.consulConfig,
+            instance.name,
```

Comment:
If you have resolvable hostnames, openstack can resolve IP based on "name". So, I think yes. We don't have resolvable hostname right now, so we're putting in a hack to get it to resolve the same way I'd assume GCP does it.